projects/packages/plugin-deactivation/src/class-deactivation-handler.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Intercept deactivation of plugins.
+ *
+ * @package automattic/jetpack-plugin-deactivation
+ */
+
+namespace Automattic\Jetpack\Plugin_Deactivation;
+
+/**
+ * Handles plugin deactivation.
+ *
+ * Instantiates the deactivation handler to intercept the deactivation of plugins.
+ */
+class Deactivation_Handler {
+
+	/**
+	 * Plugin Deactivation package version
+	 *
+	 * @var string
+	 */
+	const PACKAGE_VERSION = '0.1.0-alpha';
+
+	/**
+	 * Slug of the plugin to intercept deactivation for.
+	 *
+	 * @var string
+	 */
+	private $plugin;
+
+	/**
+	 * Feedback form url to redicret to, on deactivation.
+	 *
+	 * @var string
+	 */
+	private $feedback_url;
+
+	/**
+	 * Path to a PHP file that will be used as a template for the deactivation dialog.
+	 *
+	 * Copy ./dialog-template.php to your plugin and modify it to suit your needs.
+	 *
+	 * @var string
+	 */
+	private $dialog_view;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $plugin       Slug of the plugin to intercept deactivation for.
+	 * @param string $feedback_url Feedback form url to redicret to, on deactivation.
+	 * @param string $dialog_view  Path to a PHP file that will be used as a template for the deactivation dialog.
+	 */
+	public function __construct( $plugin, $feedback_url, $dialog_view ) {
+		$this->plugin       = $plugin;
+		$this->feedback_url = $feedback_url;
+		$this->dialog_view  = $dialog_view;
+
+		add_action( 'load-plugins.php', array( $this, 'enqueue_script' ) );
+		add_action( 'admin_footer-plugins.php', array( $this, 'embed_dialog' ) );
```

Comment:
Don't add actions to constructors. At the very least, they should be moved to `init`.

Reply:
This has been done.